tests/elements-regression/tests/regression_with_templates.test.js

```diff
@@ -0,0 +1,37 @@
+import { expect, test } from '@playwright/test';
+import _path from 'path';
+import WpAdminPage from '../src/pages/wp-admin-page';
+import EditorPage from '../src/pages/editor-page';
+
+test.describe( 'Elementor regression tests with templates for CORE', () => {
+	test.beforeAll( async ( { browser } ) => {
+		const context = await browser.newContext();
+		const page = await context.newPage();
+		const wpAdmin = new WpAdminPage( page );
+		await wpAdmin.setExperiments( {
+			container: 'active',
+		} );
+	} );
+
+	const testData = [ 'divider', 'heading', 'text_editor' ];
+	for ( const widgetType of testData ) {
+		test( `Test ${ widgetType } template`, async ( { page } ) => {
+			const filePath = _path.resolve( __dirname, `./templates/${ widgetType }.json` );
+			const wpAdminPage = new WpAdminPage( page );
+			const editorPage = new EditorPage( page );
+
+			await editorPage.setupElementorPage( wpAdminPage );
+			await editorPage.loadTemplate( filePath );
+			await editorPage.setElementorEditorPanel( 'Preview' );
+			const widgetCount = await editorPage.getWidgetCount();
+			for ( let i = 0; i < widgetCount; i++ ) {
+				await expect( editorPage.getWidget().nth( i ) ).toHaveScreenshot( `${ widgetType }_${ i }.png` );
+			}
+			await editorPage.setElementorEditorPanel( 'Back to Editor' );
+			await editorPage.publishAndViewPage();
+			await page.waitForTimeout( 500 );
+			await expect( editorPage.container ).toHaveScreenshot( `${ widgetType }_published.png`, { maxDiffPixels: 100 } );
```

Comment:
Can we do something similar to what you did here?
```typescript
			for ( let i = 0; i < widgetCount; i++ ) {
				await expect( editorPage.getWidget().nth( i ) ).toHaveScreenshot( `${ widgetType }_${ i }.png` );
			}
```

Reply:
As we discussed in slack, before publish it makes sense to get screenshot by widget. But after publish we can check everything with just one. The important thing is to do it in one container